File: Python/Leetcode Easy/303RangeSumQueryImmutable.py
#https://leetcode.com/problems/range-sum-query-immutable/description/
class NumArray:
    def __init__(self, nums: List[int]):
        self.tabArr = [0 for i in range(len(nums)+1)]
        for i in range(1, len(nums)+1):
            self.tabArr[i] = self.tabArr[i-1] + nums[i-1]

    def sumRange(self, left: int, right: int) -> int:
        return self.tabArr[right+1]-self.tabArr[left]

File: C++/Leetcode Easy/303RangeSumQuery-Immutable.cpp
#include <vector>
using namespace std;
class NumArray {
public:
    vector<int> numbers;
    vector<int> preFixSum;
    NumArray(vector<int>& nums) {
        numbers = nums;
        preFixSum.push_back(nums[0]);
        for (int i = 1; i < nums.size(); ++i) {
            preFixSum.push_back(preFixSum[i-1] + numbers[i]);
        }
    }
    
    int sumRange(int left, int right) {
        if (left == 0) return preFixSum[right];
        return preFixSum[right]-preFixSum[left-1];
    }
};

/**
 * Your NumArray object will be instantiated and called as such:
 * NumArray* obj = new NumArray(nums);
 * int param_1 = obj->sumRange(left,right);
 */